lib/madwf_transfer.cu

```diff
@@ -0,0 +1,97 @@
+#include <color_spinor_field.h>
+#include <device_vector.h>
+#include <tunable_nd.h>
+#include <madwf_transfer.h>
+#include <kernels/madwf_transfer.cuh>
+
+namespace quda
+{
+  namespace madwf_ml
+  {
+
+    template <class storage_type, class matrix_type> class transfer_5D_wrapper : public TunableKernel3D
+    {
+      ColorSpinorField &out;
+      const ColorSpinorField &in;
+      const matrix_type *wm_p;
+      bool dagger;
+
+    private:
+      unsigned int sharedBytesPerThread() const { return 0; }
+
+      unsigned int sharedBytesPerBlock(const TuneParam &) const { return out.X(4) * in.X(4) * sizeof(matrix_type); }
+
+      unsigned int minThreads() const { return out.VolumeCB() / out.X(4); }
+
+    public:
+      transfer_5D_wrapper(ColorSpinorField &out, const ColorSpinorField &in, const matrix_type *wm_p, bool dagger) :
+        TunableKernel3D(out, out.X(4), out.SiteSubset()), out(out), in(in), wm_p(wm_p), dagger(dagger)
+      {
+        TunableKernel2D_base<false>::resizeStep(out.X(4)); // Ls must be contained in the block
+
+        strcpy(aux, out.AuxString());
```

Comment:
This line should be deleted, since the parent `TunableKernel` class will already initialize the `aux` string with `out.AuxString()`